### ISSUE TITLE
Confusing error from JSON formatter when Background contains no steps

### ISSUE THREAD

**Issue by scottadav** [cucumber/cucumber-js#310]
Given the following feature file (illustrate.feature):
```
Feature: Illustrate the cucumber-js defect

Background: Something

Scenario: Illustrate defect
   Given Some step def
```

When I try to run this with the JSON formatter, it gives a very confusing error with a long stack trace.

```
$ node_modules/.bin/cucumber-js -f json -t @illustrate

/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:55
          callback();
          ^
TypeError: undefined is not a function
  at iterate (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:55:11)
  at Object.forEach (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:66:7)
  at handleBackgroundEvent (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/listener/json_formatter.js:103:11)
  at Object.hear (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/listener.js:6:9)
  at /dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:179:52
  at processItem (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:61:9)
  at iterate (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:53:11)
  at Object.forEach (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:66:7)
  at broadcastToListeners (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:178:19)
  at Object.broadcastEvent (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:184:7)
  at Object.visitBackground (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:51:12)
  at Object.instructVisitorToVisitBackground (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/ast/feature.js:97:17)
  at Object.acceptVisitor (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/ast/feature.js:89:12)
  at /dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:43:39
  at callUserFunctionAndBroadcastAfterEvent (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:153:9)
  at iterate (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:55:11)
  at Object.forEach (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:66:7)
  at broadcastToListeners (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:178:19)
  at onRuntimeListenersComplete (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:174:9)
  at iterate (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:55:11)
  at /dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:62:11
  at handleBeforeFeatureEvent (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/listener/json_formatter.js:90:5)
  at Object.hear (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/listener.js:6:9)
  at /dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:179:52
  at processItem (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:61:9)
  at iterate (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:53:11)
  at Object.forEach (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:66:7)
  at broadcastToListeners (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:178:19)
  at Object.broadcastEvent (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:184:7)
  at Object.broadcastBeforeEvent (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:163:12)
  at Object.broadcastEventAroundUserFunction (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:147:12)
  at Object.visitFeature (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:41:12)
  at /dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/ast/features.js:21:17
  at processItem (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:61:9)
  at iterate (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:53:11)
  at Object.forEach (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:66:7)
  at Object.acceptVisitor (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/ast/features.js:20:16)
  at /dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:33:40
  at callUserFunctionAndBroadcastAfterEvent (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:153:9)
  at iterate (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:55:11)
  at Object.forEach (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:66:7)
  at broadcastToListeners (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:178:19)
  at onRuntimeListenersComplete (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:174:9)
  at iterate (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:55:11)
  at /dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:62:11
  at Object.hear (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/listener.js:8:9)
  at /dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:179:52
  at processItem (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:61:9)
  at iterate (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:53:11)
  at Object.forEach (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/type/collection.js:66:7)
  at broadcastToListeners (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:178:19)
  at Object.broadcastEvent (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:184:7)
  at Object.broadcastBeforeEvent (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:163:12)
  at Object.broadcastEventAroundUserFunction (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:147:12)
  at Object.visitFeatures (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:31:12)
  at Object.walk (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime/ast_tree_walker.js:20:12)
  at Object.start (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/runtime.js:19:21)
  at Object.runSuiteWithConfiguration (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/cli.js:19:15)
  at Object.run (/dummy_cucumber_js_project/node_modules/cucumber/lib/cucumber/cli.js:12:14)
  at Object.<anonymous> (/dummy_cucumber_js_project/node_modules/cucumber/bin/cucumber.js:4:5)
  at Module._compile (module.js:456:26)
  at Object.Module._extensions..js (module.js:474:10)
  at Module.load (module.js:356:32)
  at Function.Module._load (module.js:312:12)
  at Function.Module.runMain (module.js:497:10)
  at startup (node.js:119:16)
  at node.js:901:3
```

When run with the pretty formatter, it does not have any error:
```
$ node_modules/.bin/cucumber-js -f pretty

Feature: Illustrate the cucumber-js defect



  Scenario: Illustrate defect   # features/illustrate.feature:5
    Given Some step def         # features/illustrate.feature:6


1 scenario (1 undefined)
1 step (1 undefined)

You can implement step definitions for undefined steps with these snippets:

this.Given(/^Some step def$/, function (callback) {
  // Write code here that turns the phrase above into concrete actions
  callback.pending();
});
```

**Comment by grofit** [comment]
I *think* I am getting the same issue, I am revisiting an older project and I leave the "BeforeFeatures" event handler fine and call the callback, but I never get to the "Before" handler with an error listed like yours above, and it seems to list the error origin being within the iterate method around the same number. Using the latest version 0.4.8.

**Comment by charlierudolph** [comment]
Related to #374 
Fixed in d71b92369cf72ddb2c42ae08bd76da140dcbe5bf as the wrong function was being used.
